src/shared/components/NotFound.tsx

```diff
@@ -142,25 +144,44 @@ const NotFoundOld: FC = () => (
 )
 
 class NotFound extends Component<Props> {
-  componentDidMount() {
+  state = {
+    isFetchingOrg: false,
+  }
```

Comment:
Just wondering, is this the same as setting the state in the `constructor()`. I was wondering if re-rendering the component will re-write the state in this case (?) 

This could be a confusion on my part as I am not totally sure how the lifecycle methods work.

Reply:
@ChitlangeSahas good question.

>Just wondering, is this the same as setting the state in the `constructor()`

Yep, it's the same. It's just shorthand for setting state when the component is constructed if you don't need access to `props`.

>By using an alternative class syntax, you can leave out the constructor and initialize the state as class field declaration. However, you don't have access to the `props` anymore.

From this blog post: https://www.robinwieruch.de/react-state-without-constructor/


>I was wondering if re-rendering the component will re-write the state in this case (?)

Nope, it won't re-write the state. The only thing that will rerun this code block is the component be reinitialized (not remounted) by react. I actually don't know the complete internals of react, but I believe a constructor is only called once by react, whereas `componentDidMount` can be called multiple times.

Reply:
Got it! Thank you for clarifying!!